File: bdstocktrend-backend/src/main/java/iit/nstu/bdstocktrendbackend/dto/UserDTO.java
package iit.nstu.bdstocktrendbackend.dto;

import iit.nstu.bdstocktrendbackend.models.Role;
import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
public class UserDTO {

    private Long id;

    private String username;

    private String email;

    private String name;

    private String mobile;
    
    private String photoUrl;

    private Set<String> roles = new HashSet<>();
}

File: bdstocktrend-app/lib/features/auth/data/models/user_dto.dart
class UserDto {
  final int? id;
  final String? username;
  final String? email;
  final String? name;
  final String? mobile;
  final String? photoUrl;
  final List<String>? roles;

  UserDto({
    this.id,
    this.username,
    this.email,
    this.name,
    this.mobile,
    this.photoUrl,
    this.roles,
  });

  factory UserDto.fromJson(Map<String, dynamic> json) {
    return UserDto(
      id: json['id'] as int?,
      username: json['username'] as String?,
      email: json['email'] as String?,
      name: json['name'] as String?,
      mobile: json['mobile'] as String?,
      photoUrl: json['photoUrl'] as String?,
      roles: json['roles'] != null
          ? List<String>.from(json['roles'] as List)
          : null,
    );
  }
}
